dispatches/case_studies/renewables_case/RE_surrogate_optimization_steadystate.py

```diff
@@ -0,0 +1,341 @@
+# copyright (c) 2020-2023 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#################################################################################
+
+# conceptual_design_problem_dynamic formation 2, only use timeseries clustering to cluster dispatch data
+# NN is based on dispatch_shuffled_data_0.csv, 32 clusters including capacity factor 0/1 days.
+# use omlt v1.0 in this file.
+
+#the rankine cycle is a directory above this one, so modify path
+import json
+from functools import partial
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from pathlib import Path
+import pickle
+import pyomo.environ as pyo
+from pyomo.environ import ConcreteModel, SolverFactory, units, Var, \
+    TransformationFactory, value, Block, Expression, Constraint, Param, \
+    Objective, NonNegativeReals, ConstraintList, Set, units as pyunits, RangeSet
+from pyomo.util.infeasible import log_infeasible_constraints, log_infeasible_bounds, log_close_to_bounds
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import omlt
+from omlt.neuralnet import NetworkDefinition, FullSpaceNNFormulation
+from omlt.io import load_keras_sequential
+import idaes.logger as idaeslog
+from idaes.core.solvers.get_solver import get_solver
+from idaes.apps.grid_integration.multiperiod.multiperiod import MultiPeriodModel
+
+from dispatches.case_studies.renewables_case.load_parameters import pem_bar, pem_temp, wind_cap_cost, \
+                                wind_op_cost, pem_op_cost, pem_var_cost, pem_cap_cost, PA, h2_mols_per_kg
+from dispatches.case_studies.renewables_case.wind_battery_PEM_LMP import wind_battery_pem_variable_pairs, \
+                                wind_battery_pem_periodic_variable_pairs, wind_battery_pem_om_costs, \
+                                initialize_mp, wind_battery_pem_model, wind_battery_pem_mp_block
+
+
+# path for folder that has surrogate models
+re_nn_dir = Path(__file__).parent / "data" / "steady_state_surrogate"
+
+def load_surrogate_model(re_nn_dir):
+    # load scaling and bounds for NN surrogates (rev and # of startups)
+    with open(re_nn_dir / "dispatch_frequency" / "static_clustering_wind_pmax.pkl", 'rb') as f:
+        model = pickle.load(f)
+    centers = model.cluster_centers_
+    dispatch_clusters_mean = centers[:, 0]
+    pem_clusters_mean = centers[:, 1]
+    resource_clusters_mean = centers[:, 2]
+
+    with open(re_nn_dir / "revenue" / "RE_revenue_params_2_25.json", 'rb') as f:
```

Comment:
```suggestion
    with open(re_nn_dir / "revenue" / "RE_revenue_params_2_25.json", 'r') as f:
```

Not sure -- it would only matter for Windows. We use `'r'` for a json below.

Reply:
fixed